File: Python/TwoSum.py
# -*- coding: utf-8 -*-
"""
Created on Thu Nov  2 00:43:42 2017

@author: Ankit
"""

def twoSum(nums,target):
    diffDict = {};
    for i in range(len(nums)):
        diff = target - nums[i];
        print (diff);
        if nums[i] in diffDict:
            return [diffDict[nums[i]],i];
        diffDict[diff] = i;
    print(diffDict);
    return []

File: Java/TwoSum.java
import java.util.Arrays;


public class TwoSum {
	public static void main ( String args[] ){
		int nums[] = {2,5,5,15};
		int target = 10;
		int[] out = new int[2];
        for ( int i =0 ; i< nums.length ; i++ ){
            for ( int j = i+1 ; j< nums.length ;j++){
                if ( (nums[i] + nums[j]) == target ){
                    out[0] = i;
                    out[1] = j;
                }
             }
        }
        System.out.println("Out : "+Arrays.toString(out));
	}
}
